numba/jitclass/base.py

```diff
@@ -104,6 +113,19 @@ def _validate_spec(spec):
             raise TypeError("spec values should be Numba type instances, got %r"
                             % (v,))
 
+
+def _fix_up_private_attr(clsname, spec):
+    """
+    Apply the same changes to dunder names as CPython would.
+    """
+    out = OrderedDict()
+    for k, v in spec.items():
+        if k.startswith('__'):
```

Comment:
I think you want to guard for `k.endswith('__')`, as `__special__` methods are not mangled.